bundles/org.openhab.binding.mqtt.espmilighthub/src/main/java/org/openhab/binding/mqtt/espmilighthub/internal/handler/EspMilightHubHandler.java

```diff
@@ -0,0 +1,383 @@
+                }
+                String bulbDiscoMode = Helper.resolveJSON(messageJSON, "\"mode\":", 1);
+                if (!bulbDiscoMode.isEmpty()) {
+                    changeChannel(CHANNEL_DISCO_MODE, new StringType(bulbDiscoMode));
+                }
+                break;
+            case "night":
+                if (!globeType.equals("cct") && !globeType.equals("fut091")) {
+                    changeChannel(CHANNEL_BULB_MODE, new StringType("night"));
+                    if (config.oneTriggersNightMode) {
+                        changeChannel(CHANNEL_LEVEL, new PercentType("1"));
+                    }
+                }
+                break;
+        }
+    }
+
+    /*
+     * Used to calculate the colour temp for a globe if you want the light to get warmer as it is dimmed like a
+     * traditional halogen globe
+     */
+    private int autoColourTemp(int brightness) {
+        return minColourTemp.subtract(
+                minColourTemp.subtract(maxColourTemp).divide(new BigDecimal(100)).multiply(new BigDecimal(brightness)))
+                .intValue();
+    }
+
+    void turnOff() {
+        if (config.powerFailsToMinimum) {
+            sendMQTT("{\"state\":\"OFF\",\"level\":0}");
+        } else {
+            sendMQTT("{\"state\":\"OFF\"}");
+        }
+    }
+
+    void handleLevelColour(Command command) {
+        if (command instanceof OnOffType) {
+            if (OnOffType.ON.equals(command)) {
+                sendMQTT("{\"state\":\"ON\",\"level\":" + savedLevel + "}");
+                return;
+            } else {
+                turnOff();
+            }
+        } else if (command instanceof IncreaseDecreaseType) {
+            if (IncreaseDecreaseType.INCREASE.equals(command)) {
+                if (savedLevel.intValue() <= 90) {
+                    savedLevel = savedLevel.add(new BigDecimal(10));
+                }
+            } else {
+                if (savedLevel.intValue() >= 10) {
+                    savedLevel = savedLevel.subtract(new BigDecimal(10));
+                }
+            }
+            sendMQTT("{\"state\":\"ON\",\"level\":" + savedLevel.intValue() + "}");
+            return;
+        } else if (command instanceof HSBType) {
+            HSBType hsb = (HSBType) command;
+            // This feature allows google home or Echo to trigger white mode when asked to turn color to white.
+            if (hsb.getHue().intValue() == config.whiteHue && hsb.getSaturation().intValue() == config.whiteSat) {
+                if ("rgb_cct".equals(globeType) || "fut089".equals(globeType)) {
+                    sendMQTT("{\"state\":\"ON\",\"color_temp\":" + config.favouriteWhite + "}");
+                } else {// globe must only have 1 type of white
+                    sendMQTT("{\"command\":\"set_white\"}");
+                }
+                return;
+            } else if (hsb.getBrightness() == PercentType.ZERO) {
```

Comment:
`==` shouldn't be used in this case since PercentType isn't a primitive, multiton, or enum.
```suggestion
            } else if (PercentType.ZERO.equals(hsb.getBrightness())) {
```